src/ast/ast.ts

```diff
@@ -547,6 +549,13 @@ export type AstBoolean = {
     loc: SrcInfo;
 };
 
+// Reference to source file
+export type SourceReference = {
+    readonly path: RelativePath;
+    readonly type: "stdlib" | "relative";
```

Comment:
1. Maybe it should be

```ts
readonly root: "stdlib" | "project";
```

?

2. It also feels like an overlap with `ItemOrigin`.